app.js

```diff
@@ -355,6 +389,11 @@ function renderPage( type, options ) {
   var query     = options.query;
   var debug     = options.debug;
   var cssCookie = options.cssCookie;
+  var isTools   = type === 'tools';
+
+  var isDemoToolAdded = isTools ? _.some( list, function( tool ) {
+    return tool.name === '_DEMO TOOL_';
+  } ) : false;
 
```

Comment:
I know it feels wrong, but I really tried to think of the best possible way to prevent multiple useless iteration over the data collections. Anyway I will move it to the dedicated function and will test again. Thanks!